### PR TITLE
feat(docker): Dockerfile and README update for basic docker development workflow

### PR DESCRIPTION
**Pull request by jamonation**
I've added a dockerfile and some instructions to README.md explaining how to use Docker to develop and run the content server. I've also set it up so that the container runs Firefox with Selenium for testing. It allows for development with an IDE underneath a running container as one would normally work. This means changes and running the verifier are mostly transparent to normal development. If there are any questions feel free to ask here or in IRC.

### CODE REVIEW

**Comment by zaach** [comment]
Neat. Is this part of a larger docker setup you've got?

**Comment by shane-tomlinson** [comment]
Giving this a spin.

> Neat. Is this part of a larger docker setup you've got?

@zaach - yeah, @jamonation has been adding PRs to all of our repos, one by one. :+1:

**Comment by jamonation** [comment]
@zaach @shane-tomlinson, @vladikoff and I chatted yesterday about this. I showed all the components working together with one Docker based synchronization/coordination tool (fig). It is like pm2 in @vladikoff's fxa-dev repository, but being Docker means that Windows, OSX, and Linux users would have the exact same setup. 

Right now I'm looking at how to cache npm modules during builds. Something like npm-cache should make full setups of the whole dev stack easier for those with bandwidth constraints. More to come!

File: README.md [hunk 1]
```diff
@@ -30,6 +30,18 @@ It will listen on <http://127.0.0.1:3030> by default.
 Note: If you have issues with `npm install` please update to npm 2.0+ using `npm install -g npm@2` 
 ([Issue #1594](https://github.com/mozilla/fxa-content-server/issues/1594))
 
+## Docker Based Development
+
+To run the content server via Docker, three steps are required:
+
+    $ docker build --rm -t mozilla/fxa_content_server
```
Comment (shane-tomlinson) [inline README.md]: When I ran this command locally, from the fxa-content-server directory, I had to append a `.`. This is on OSX using boot2docker/docker 1.5.0

File: README.md [hunk 2]
```diff
@@ -30,6 +30,18 @@ It will listen on <http://127.0.0.1:3030> by default.
...
+    $ docker run --rm -v $PWD:/opt/fxa mozilla/fxa_content_server npm install
+    $ docker run --rm -v $PWD:/opt/fxa --net=host mozilla/fxa_content_server
+
+This method shares the codebase into the running container so that you can install npm and various modules required by package.json. It then runs FxA content server in a container, while allowing you to use your IDE of choice from your normal desktop environment to develop code.
+
+Be sure to copy server/config/local.json-dist to server/config/local.json per usual before the final docker invocation to run the service.
```
Comment (shane-tomlinson) [inline README.md]: Being new to docker, one thing I found strange is I had to access the content server via the ip address reported by:

```
> boot2docker ip
192.168.59.103
```

This means the config.json needs to be updated, as well as the [contentServer.url config item](https://github.com/mozilla/fxa-auth-server/blob/d5ae3b59cc1c7add09a3e70edfbe70b0f0321f74/config/dev.json#L3) in the fxa-auth-server.

Maybe make a note of this, or else show some instructions to newbies like me how I can access the docker container via localhost/127.0.0.1

Comment (jamonation) [inline README.md]: Thanks for pointing that out, I'll incorporate that into the readme (for this and the other repos) and update the various pull requests. The fully synchronized version that I mentioned will handle these things in a more consistent way.

File: README.md [hunk 3]
```diff
@@ -30,6 +30,18 @@ It will listen on <http://127.0.0.1:3030> by default.
...
+
```
Comment (shane-tomlinson) [inline README.md]: Info on stopping the container and restarting the server would also be useful for n00bs like me.

File: README.md [hunk 4]
```diff
@@ -58,6 +70,22 @@ To run tests locally with Selenium:
...
+To run tests in Docker with Selenium, first run:
+
+```
+docker run -i -t --rm -v $PWD:/opt/fxa --net=host mozilla/fxa_content_server /bin/bash
+```
+
+You will now be inside a container, wherein you can run tests. Be sure to have downloaded the selenium jar file beforehand, then:
+
+``sh
+java -jar selenium-server-standalone-2.43.1.jar &
```
Comment (shane-tomlinson) [inline README.md]: It's worth noting that 2.44.0 does *not* work inside the container.

**Comment by shane-tomlinson** [comment]
@jamonation - this is really some fantastic work, thank you for this! I'm up for merging this if you can make a few updates.

**Comment by jamonation** [comment]
I've poked at this a bit more and it turns out no selenium tests work inside docker. I've tried running Firefox with Xvfb, and though it works on its own, and selenium works on its own in Docker, getting both going just seems to be a non-starter.

I'll update the readme noting this later today.

**Comment by jamonation** [comment]
Tidied up and ready for review again.

**Pull request merged**